app/src/androidTest/java/com/nervousfish/nervousfish/test/ContactActivitySteps.java

```diff
@@ -46,16 +64,21 @@ public ContactActivitySteps(EntryActivity activityClass) {
     }
 
     @Given("^I am viewing the contact activity$")
-    public void iAmViewingContactActivity() {
+    public void iAmViewingContactActivity() throws IOException {
         assertNotNull(getActivity());
 
-        IKey key = new SimpleKey("my key", "key");
-        Contact contact = new Contact("Cornel",key );
+        contact = new Contact(TEST_NAME, TEST_KEY);
+
+        serviceLocator = (IServiceLocator) getCurrentActivity().getIntent().getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+        List<Contact> contacts = serviceLocator.getDatabase().getAllContacts();
+        for (Contact contact: contacts) {
+            serviceLocator.getDatabase().deleteContact(contact.getName());
+        }
+        serviceLocator.getDatabase().addContact(contact);
```

Comment:
I think it would be better here to mock the database because you shouldn't (1) rely on the current contents of the database and (2) alter the contents of the database. You can do this by given the `ContactActivity` an intent with a mocked `ServiceLocator`